### PR TITLE
Ember Blueprint Updates

### PR DESCRIPTION
**Pull request by elwayman02**
Test specifically against Node 20 now that Node 16 is EOL

### CODE REVIEW

**Comment by SergeAstapov** [comment]
@elwayman02 it's not breaking as v2 addons do not specify Node.js version they depend upon

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,17 +45,15 @@ jobs:
...
     runs-on: ubuntu-latest
-    needs: "test"
+    needs: 'test'
     timeout-minutes: 10
 
     strategy:
       fail-fast: false
       matrix:
         try-scenario:
-          - ember-lts-3.28
```
Comment (SergeAstapov) [inline .github/workflows/ci.yml:55]: maybe we can keep 3.28+ versions in CI and then it's not breaking change?

Comment (elwayman02) [reply .github/workflows/ci.yml:55]: I was planning to bump ember-resolver to latest, which does break below 4.12

Comment (elwayman02) [reply .github/workflows/ci.yml:55]: That said, if there's a strong reason we need to land some of these changes as a non-major release, we could break it into multiple PRs and do a minor release first.

Comment (SergeAstapov) [reply .github/workflows/ci.yml:55]: @elwayman02 there is no argument why we should not be doing major.

however, at the same time there are technically no breaking changes, e.g. `ember-resolver` is just a dev dependency and in other addonsI I was downgrading it in ember-try to v9 for <= v4.8 scenarios which allows green CI without any major bumps.

just sharing experience, if you interested you may peek into https://github.com/ember-modifier/ember-modifier/blob/main/test-app/config/ember-try.js#L51

Comment (elwayman02) [reply .github/workflows/ci.yml]: That's a fair point. I went ahead and added a resolution on v11 for those scenarios and re-added them.

**Comment by elwayman02** [comment]
Ok, this is mostly ready except for a weird linter parsing error in the rollup config. Once we figure that out, we can see if tests pass and get this out the door.

File: addon/.eslintrc.cjs [hunk 2]
```diff
@@ -2,29 +2,76 @@
...
-  ],
-  env: {
-    browser: true,
-  },
-  rules: {},
+  // Only use overrides
+  // https://github.com/ember-cli/eslint-plugin-ember?tab=readme-ov-file#gtsgjs
   overrides: [
+    {
+      files: ['**/*.js', '**/*.ts'],
```
Comment (SergeAstapov) [inline addon/.eslintrc.cjs:9]: @elwayman02 I think this should be
```suggestion
      files: ['src/**/*.js', 'src/**/*.ts'],
```
to fix that pesky `Parsing error: 'import' and 'export' may appear only with 'sourceType: module'`

I'll check later if it's issue in blueprint and submit PR upstream

Comment (elwayman02) [reply addon/.eslintrc.cjs:9]: Looks like the issue was the node file config not being setup to handle `.mjs` files...turns out if I remove the rollup file from that config for now, everything passes. There's a more correct answer (fixing the config to parse mjs properly), but this unblocks me and the blueprint will need that fix anyway so I'll pick it up eventually here.

Comment (SergeAstapov) [reply addon/.eslintrc.cjs:9]: @elwayman02 ahha, ok, interesting. We need to fix this up separately in the blueprint. Thanks for digging in!

**Pull request merged**